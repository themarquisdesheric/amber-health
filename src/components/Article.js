import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'gatsby';
import PreviewCompatibleImage from './PreviewCompatibleImage';

const Article = ({ post, className = '' }) => (
  <article className={`${className} mb-8`}>
    <Link to={post.fields.slug}>
      <header>
        {post.frontmatter.featuredimage ? (
          <div className="w-full">
            <PreviewCompatibleImage
              imageInfo={{
                image: post.frontmatter.featuredimage,
                alt: `featured image thumbnail for post ${post.title}`
              }}
            />
          </div>
        ) : null}
        <p className="text-lg font-bold pt-5 pb-2">
          {post.frontmatter.title}
        </p>
      </header>
      <p className="text-sm pb-2">{post.frontmatter.description}</p>
    </Link>
  </article>
);

Article.propTypes = {
  post: PropTypes.object.isRequired,
  className: PropTypes.string
};

export default Article;
