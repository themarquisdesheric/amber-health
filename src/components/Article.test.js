import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Article from './Article';

const post = {
  id: 'a3a3d5f0-fd76-5738-9d6c-b6e250ee9f10',
  fields: {
    slug: '/articles/2019-09-27-endometriosis-diagnosis/'
  },
  frontmatter: {
    title: 'Endometriosis Diagnosis',
    description: 'Learn how endometriosis is diagnosed',
    draft: false,
    templateKey: 'article',
    featuredimage: {
      childImageSharp: {
        fluid: {
          base64: 'base64',
          aspectRatio: 1.5,
          src: '/static/a162fe840d8c214c93ae37d3e6fc212a/6806c/pom-2.jpg',
          srcSet: 'srcSet',
          sizes: '(max-width: 576px) 100vw, 576px'
        }
      }
    }
  }
};

test('Article', () => {
  const wrapper = shallow(<Article post={post} index={0} />);

  expect(toJson(wrapper)).toMatchSnapshot();
});
