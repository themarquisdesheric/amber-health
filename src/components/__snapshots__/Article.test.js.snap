// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article 1`] = `
<article
  className=" mb-8"
>
  <mockConstructor
    to="/articles/2019-09-27-endometriosis-diagnosis/"
  >
    <header>
      <div
        className="w-full"
      >
        <PreviewCompatibleImage
          imageInfo={
            Object {
              "alt": "featured image thumbnail for post Endometriosis Diagnosis",
              "image": Object {
                "childImageSharp": Object {
                  "fluid": Object {
                    "aspectRatio": 1.5,
                    "base64": "base64",
                    "sizes": "(max-width: 576px) 100vw, 576px",
                    "src": "/static/a162fe840d8c214c93ae37d3e6fc212a/6806c/pom-2.jpg",
                    "srcSet": "srcSet",
                  },
                },
              },
            }
          }
        />
      </div>
      <p
        className="text-lg font-bold pt-5 pb-2"
      >
        Endometriosis Diagnosis
      </p>
    </header>
    <p
      className="text-sm pb-2"
    >
      Learn how endometriosis is diagnosed
    </p>
  </mockConstructor>
</article>
`;
